Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/worker/RatResultRetrievalWorker.kt

```diff
@@ -0,0 +1,94 @@
+package de.rki.coronawarnapp.coronatest.worker
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.coronatest.CoronaTestRepository
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.DISABLED
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.PHASE1
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.PHASE2
+import de.rki.coronawarnapp.coronatest.latestRAT
+import de.rki.coronawarnapp.coronatest.type.CoronaTest
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.worker.InjectedWorkerFactory
+import de.rki.coronawarnapp.worker.BackgroundConstants
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import timber.log.Timber
+
+/**
+ * Diagnosis test result retrieval by periodic polling
+ */
+class RatResultRetrievalWorker @AssistedInject constructor(
```

Comment:
Renaming the workers breaks the already scheduled workers as they are started based on their class name.
I think we should still rename, as in any case we need a "rescheduling" mechanism.
This comment as reminder, we'll think about a solution toegether.